src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -64,6 +73,110 @@ def create_resource_group(rg_name, location, tags=None):
+            raise CLIError('--plan-name, --plan-product, --plan-publisher and --plan-version are all required if kind is ServiceCatalog')
+        else:
+            appliancePlan = Plan(plan_name, plan_publisher, plan_product, plan_version)
+            appliance.plan = appliancePlan
+
+    applianceParameters = None
+
+    if parameters:
+        if os.path.exists(parameters):
+            applianceParameters = get_file_json(parameters)
+        else:
+            applianceParameters = shell_safe_json_parse(parameters)
+
+    applianceProperties.parameters = applianceParameters
+
+    return racf.appliances.create_or_update(resource_group_name, appliance_name, appliance)
+
+def show_appliance(resource_group_name=None, appliance_name=None, appliance_id=None):
+    ''' Gets an appliance.
+    :param str resource_group_name:the resource group name
+    :param str appliance_name:the appliance name
+    '''
+    racf = _resource_appliances_client_factory()
+    if appliance_id:
+        appliance = racf.appliances.get_by_id(appliance_id)
+    else:
+        appliance = racf.appliances.get(resource_group_name, appliance_name)
+    return appliance
+
+def show_appliancedefinition(resource_group_name=None, appliance_definition_name=None, appliance_definition_id=None):
+    ''' Gets an appliance definition.
+    :param str resource_group_name:the resource group name
+    :param str appliance_definition_name:the appliance definition name
+    '''
+    racf = _resource_appliances_client_factory()
+    if appliance_definition_id:
+        appliancedef = racf.appliance_definitions.get_by_id(appliance_definition_id)
+    else:
+        appliancedef = racf.appliance_definitions.get(resource_group_name, appliance_definition_name)
+    return appliancedef
+
+def create_appliancedefinition(resource_group_name, appliance_definition_name, location, lock_level, package_file_uri, authorizations, description, display_name, tags=None):
+    ''' Create a new appliance definition.
+    :param str resource_group_name:the desired resource group name
+    :param str appliance_definition_name:the appliance definition name
+    :param str description:the appliance definition description
+    :param str display_name:the appliance definition display name
+    :param str package_file_uri:the appliance definition package file uri
+    :param str tags:tags in 'a=b c' format
+    '''
+    racf = _resource_appliances_client_factory()
+    authorizations = authorizations or []
+    applianceAuth = []
+
+
+    for name_value in authorizations:
+        # split at the first ':', both principalId and roldeDefinitionId should not have a ':'
+        principalId, roleDefinitionId = name_value.split(':', 1)
+        applianceAuth1 = ApplianceProviderAuthorization(principalId, roleDefinitionId)
+        applianceAuth.append(applianceAuth1)
+    applianceDefProperties = ApplianceDefinitionProperties(lock_level, applianceAuth, package_file_uri, display_name, None, description)
```

Comment:
I've flattened the Appliance model. We no longer have properties model for both Appliance and ApplianceDefinitions